syntax = "proto3";

package telemetry;

// One metric
message Metric {
  string name = 1;

  oneof value {
    uint64 counter_value = 2;
    int64 gauge_value = 3;
  }
}

// Batch of metrics
message MetricBatch {
  repeated Metric metrics = 1;
}

// Acknowledgement
message Ack {
  bool success = 1;
}

// Telemetry service
service TelemetryService {
  rpc PushMetrics(MetricBatch) returns (Ack);
}

